src/lib.rs

```diff
@@ -46,9 +46,14 @@ pub fn guess_js_syntax(s: &str) -> JsSyntax {
 
         // single line comment, ignore until \n
         if c == b'/' && b[i + 1] == b'/' {
-            // TODO: handle \r\n?
-            let new_line_pos = match b[i + 2..].iter().position(|&v| v == b'\n') {
-                Some(pos) => pos,
+            let new_line_pos = match b[i + 2..].iter().position(|&v| v == b'\n' || v == b'\r') {
+                Some(pos) => {
+                  if pos > 0 && b[i + 2 + pos - 1] == b'\r' {
+                    pos + 1
+                  } else {
+                    pos
+                  }
+                },
```

Comment:
Thanks for the PR! It's been a long long time since I touch Rust 😄 

IIUC with this change, if `v` matches `\n` or `\r`, it would also check before it, if it's a `\r`? In that case, wouldn't the previous code work too? Given `// foobar \r\n`, since `\n` appears after `\r`, it would iterate till `\n` then skip the `pos`.

I think instead of checking the char before it, we should check after for `\n`. So if we match `\r`, we then test for `\n`. Doing this even if we match `\n` seems fine too.

Reply:
> Given // foobar \r\n, since \n appears after \r, it would iterate till \n then skip the pos.

Ah, errr, yeah I guess you're right there lol. 😄 

> So if we match \r, we then test for \n.

But then this shouldnt be needed either right? 🤔

Reply:
Since we check `v == b'\n' || v == b'\r'`, we could match the first part of `\r\n` only, so we can check for the next one too. But I guess not doing so is fine as a new line would also be skipped in other parts of the code anyway, but it does squeeze out a little perf. I blame past me for the confusion 😬 

Feel free if you'd like to work on the perf bit.

Reply:
Oops, looks like I was actually off by one character 😅 

> I think instead of checking the char before it, we should check after for \n. So if we match \r, we then test for \n. Doing this even if we match \n seems fine too.

So the way I read this code is: 

```rust
            let new_line_pos = match b[i + 2..].iter().position(|&v| v == b'\n' || v == b'\r') {
                Some(pos) => {
                  // if the char at i + 2 + pos == '\r', we can early return and increment pos + 1 (+1 for the '\n')
                  // e.g.:
                  // b"// hello world\r\n
                  //   iippppppppppppp
                  //   ++ooooooooooooo
                  //   22sssssssssssss
                  if pos > 0 && b[i + 2 + pos] == b'\r' {
                    // its a b'\r' so we have to add + 1 to account for `\n`
                    pos + 1
                  } else {
                    // else: the item at `pos` is a b'\n', so we can return the `pos`
                    pos
                  }
                },
                None => break,
            };
```

But I might be misunderstanding your original comment lol 😄 At any rate, feel free to close this PR, I'm not sure if the perf gain is actually worth it. I mostly wanted to just try and get some hands on experience contributing to a Rust codebase.

Reply:
A lone `\r` could also be considered a new line, so we can't `+ 1` in that case. Changing `b[i + 2 + pos] == b'\r'` to `b[i + 2 + pos + 1] == b'\n'` should do the trick, so it checks the next `\n` and skip if so.

That should be good enough for the PR 🙂

Reply:
Ahh, I thought `\r` was always supposed to be followed by a `\n`, my bad. Ill adjust the PR today, thanks for the pointers!

Reply:
Pushed 🙂 And thanks for bearing with me! 😄